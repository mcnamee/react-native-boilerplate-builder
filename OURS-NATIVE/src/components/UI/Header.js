import React from 'react';
import PropTypes from 'prop-types';
import { View } from 'react-native';
import { Text, H1 } from 'native-base';
import Spacer from './Spacer';

const Header = ({ title, content }) => (
  <View>
    <Spacer size={25} />
    <H1>{title}</H1>
    {!!content && (
      <View>
        <Spacer size={10} />
        <Text>{content}</Text>
      </View>
    )}
    <Spacer size={25} />
  </View>
);

Header.propTypes = {
  title: PropTypes.string,
  content: PropTypes.string,
};

Header.defaultProps = {
  title: 'Missing title',
  content: '',
};

export default Header;
