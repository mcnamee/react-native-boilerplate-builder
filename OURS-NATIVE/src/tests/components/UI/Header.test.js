import React from 'react';
import renderer from 'react-test-renderer';
import { render } from '@testing-library/react-native';
import Header from '../../../components/UI/Header';

it('<Header /> renders with message', () => {
  const Component = <Header title="hello boy" content="I'm here" />;

  // Matches snapshot
  expect(renderer.create(Component).toJSON()).toMatchSnapshot();

  // Has the correct text on the page
  const { getByText } = render(Component);
  expect(getByText('hello boy'));
  expect(getByText("I'm here"));
});
