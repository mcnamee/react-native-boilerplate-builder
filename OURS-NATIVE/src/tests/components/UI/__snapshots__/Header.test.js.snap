// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> renders with message 1`] = `
<View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "height": 25,
        },
      ]
    }
  />
  <Text
    style={
      Object {
        "color": "#000",
        "fontSize": 27,
        "lineHeight": 32,
      }
    }
  >
    hello boy
  </Text>
  <View>
    <View
      style={
        Array [
          Object {},
          Object {
            "flex": 1,
            "height": 10,
          },
        ]
      }
    />
    <Text
      style={
        Object {
          "color": "#000",
          "fontFamily": "System",
          "fontSize": 16,
        }
      }
    >
      I'm here
    </Text>
  </View>
  <View
    style={
      Array [
        Object {},
        Object {
          "flex": 1,
          "height": 25,
        },
      ]
    }
  />
</View>
`;
